tests/Core/Tokenizer/ExplicitOctalNotationTest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Tests the tokenization of explicit octal notation to PHP < 8.1.
+ *
+ * @author    Mark Baker <mark@demon-angel.eu>
+ * @copyright 2019 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Tokenizer;
+
+use PHP_CodeSniffer\Tests\Core\AbstractMethodUnitTest;
+
+class ExplicitOctalNotationTest extends AbstractMethodUnitTest
+{
+
+
+    /**
+     * Test that explicitly-defined octal values are tokenized as a single number and not as a number and a string.
+     *
+     * @param array $testData The data required for the specific test case.
+     *
+     * @dataProvider dataExplicitOctalNotation
+     * @covers       PHP_CodeSniffer\Tokenizers\PHP::tokenize
+     *
+     * @return void
+     */
+    public function testExplicitOctalNotation($testData)
+    {
+        $tokens = self::$phpcsFile->getTokens();
+
+        $number = $this->getTargetToken($testData['marker'], [T_LNUMBER, T_DNUMBER, T_STRING]);
```

Comment:
Are we only looking for T_LNUMBER tokens here or are there test cases to add where T_DNUMBER and T_STRING would be valid to find? Might need some tests to ensure some strings that look octal are not actually being touched ?